op-service/sources/l1_beacon_client.go

```diff
@@ -129,6 +129,19 @@ func (cl *BeaconHTTPClient) BeaconBlobSideCars(ctx context.Context, fetchAllSide
 	if err := cl.apiReq(ctx, &resp, reqPath, reqQuery); err != nil {
 		return eth.APIGetBlobSidecarsResponse{}, err
 	}
+
+	indices := make(map[uint64]bool)
+	for _, h := range hashes {
+		indices[h.Index] = true
+	}
+
+	for _, apisc := range resp.Data {
+		delete(indices, uint64(apisc.Index))
+	}
+
+	if len(indices) > 0 {
+		return eth.APIGetBlobSidecarsResponse{}, fmt.Errorf("#returned blobs(%d) != #requested blobs(%d)", len(hashes)-len(indices), len(hashes))
```

Comment:
```suggestion
		return eth.APIGetBlobSidecarsResponse{}, fmt.Errorf("#returned blobs(%d) != #requested blobs(%d)", len(resp.Data), len(hashes))
```
find this easier to reason

Reply:
This will be misleading when `fetchAllSidecars`